src/vivarium_census_prl_synth_pop/components/observers.py

```diff
@@ -89,78 +89,47 @@ def on_simulation_end(self, event: Event) -> None:
-    save population table with / key = date
-    """
-
-    def __repr__(self) -> str:
-        return "Observers()"
-
-    ##############
-    # Properties #
-    ##############
+    def __repr__(self):
+        return f"CensusObserver({self.census_date_str})"
 
     @property
     def name(self):
-        return "observers"
-
-    #################
-    # Setup methods #
-    #################
+        return f"census_observer_{self.census_date_str}"
+    
+    @property
+    def output_filename(self):
+        return f"decennial_census_{self.census_date_str}.hdf"
 
     def setup(self, builder: Builder):
-        self.start_date = builder.configuration.time.start
-        self.end_date = builder.configuration.time.end
+        super().setup(builder)
         self.clock = builder.time.clock()
-        self.counter = 0
-        # FIXME: Are these the correct output locations?
-        self.output_path = build_output_dir(builder, subdir="population_table") / "state_table.hdf"
-        self.decennial_path = build_output_dir(builder, subdir="population_table") / "decennial_census.hdf"
-
-        self.randomness = builder.randomness.get_stream(self.name)
-        self.population_view = builder.population.get_view(columns=[])
-        self.response_probability_decennial = builder.lookup.build_table(
-            data=data_values.RESPONSE_PROBABILITY_DECENNIAL
-        )
+        
+    def get_population_view(self, builder) -> PopulationView:
+        """Get the population view to be used for observations"""
+        return builder.population.get_view(columns=metadata.DECENNIAL_CENSUS_COLUMNS_USED)
 
-        # FIXME: register to happen "on_collect_metrics" (end of time step)
-        builder.event.register_listener("time_step__prepare", self.on_time_step__prepare)
-        builder.event.register_listener("simulation_end", self.on_simulation_end)
-
-    def on_time_step__prepare(self, event: Event) -> None:
-        if self.counter == 0:
-            start_date_str = (
-                f"ymd_{self.start_date.year}_{self.start_date.month}_{self.start_date.day}"
-            )
-            state_table = self.population_view.get(event.index)
-            state_table.to_hdf(self.output_path, start_date_str)
-        self.counter += 1
-        if (self.clock().year % 10 == 0) & (self.clock().month == 4):
-            if self.clock().day < 29:  # because we only want one observation in April
-                self.decennial_census(event, hdf_key=f"year_{self.clock().year}")
+    def get_responses(self) -> pd.DataFrame:
+        return pd.DataFrame()
 
-    def on_simulation_end(self, event: Event) -> None:
-        end_date_str = f"ymd_{self.end_date.year}_{self.end_date.month}_{self.end_date.day}"
-        state_table = self.population_view.get(event.index)
-        state_table.to_hdf(self.output_path, end_date_str)
+    def to_observe(self, event: Event) -> bool:
+        if (self.clock().year % 10 == 0) & (self.clock().month == 4):
```

Comment:
Unless I'm missing something, as implemented, the two datasets will be identical. I think what you want is to check that `self.clock().year == self.census_date.year`

Reply:
What we actually want here is 
```
return event.time.year % 10 == 0 and event.time.month == 4 and event.time.day <= self.time_step
```
so that we observe on the first April timestep of each decade. Note: `self.time_step` should be retrieved during `setup` so that it refers to the actual time-step length used in the simulation rather than being hard-coded as below.

Reply:
But he is instantiating the obsever twice, each time with a different date string, eg
- CensusObserver('2020-04-01')
- CensusObserver('2030-04-01')

Reply:
You are correct, I plan to add improved logic for this in a future PR, by testing if the `census_date` is between `event.time` and `self.clock.time()`.  As above, I thought leaving it for a separate PR would make it simpler for code review... sorry if I misjudged that!

Reply:
Rajan strongly prefers an approach with a single observer, so that we can change the duration of the sim and get all the decennial censuses observed without having to add additional `CensusObserver` lines to the model spec, which I have implemented in e6c49f7f4e0a888f329bf362481af84b110032a0